mern-ecommerce/controllers/user.js

```diff
@@ -40,6 +41,20 @@ exports.update = (req, res) => {
   );
 };
 
+exports.productById = (req, res, next, id) => { 
+  Product.findById(id) 
+    .populate('products.product', 'name price')
+    .exec((err, product) => { 
+      if (err || !product) {
+        return res.status(400).json({
+          error: errorHandler(err),
+        });
+      }
+      req.product = product; 
```

Comment:
The variable name `req.product` should be consistent with the rest of the code, which uses `req.profile` for user data. It should be renamed to `req.productProfile` for clarity and consistency.